src/libs/migrations/CheckForPreviousReportActionID.js

```diff
@@ -0,0 +1,71 @@
+import _ from 'underscore';
+import Onyx from 'react-native-onyx';
+import Log from '../Log';
+import ONYXKEYS from '../../ONYXKEYS';
+
+/**
+ * @returns {Promise<Object>}
+ */
+function getReportActionsFromOnyx() {
+    return new Promise((resolve) => {
+        const connectionID = Onyx.connect({
+            key: ONYXKEYS.COLLECTION.REPORT_ACTIONS,
+            waitForCollectionCallback: true,
+            callback: (allReportActions) => {
+                Onyx.disconnect(connectionID);
+                return resolve(allReportActions);
+            },
+        });
+    });
+}
+
+/**
+ * Checks if the input object is not null, empty or undefined.
+ *
+ * @param {Object} data
+ * @returns {Boolean}
+ */
+function isValid(data) {
+    return data !== null && !_.isEmpty(data) && !_.isUndefined(data);
+}
+
+/**
+ * Migrate Onyx data for reportActions. If the first reportAction of a reportActionsForReport
+ * does not contain a 'previousReportActionID', all reportActions for all reports are removed from Onyx.
+ *
+ * @returns {Promise<void>}
+ */
+export default function () {
+    return getReportActionsFromOnyx().then((allReportActions) => {
+        if (!isValid(allReportActions)) {
+            Log.info(`[Migrate Onyx] Skipped migration CheckForPreviousReportActionID because there were no reportActions`);
+            return;
+        }
+
+        const firstValidReportAction = _.find(_.values(allReportActions), (reportAction) => isValid(reportAction));
```

Comment:
I don't think we need a new isValid function, we can use the lodashHas to find a report action with a `reportActionID` field. Then We can use the reportAction to find flow to the next migration step.

@neonbhai